doc/design/staging-1.md

```diff
@@ -0,0 +1,37 @@
+# Staging Environment and Procedures
+
+## Production Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Sponsored S3 bucket in the sponsored account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+## Staging Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Private S3 bucket in the project account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+This should exactly parallel production, except for the S3 bucket, which is private.
+Nothing is actually shared, everything is hosted adjacently.
+
+The staging server will contain some test data and some (smallish) real data that reflects the contents of production.
+
+## Release procedure
+The staging server will track the `master` branch in `dandi-api` and will automatically redeploy whenever `master` is changed.
+
+The production server will not track any branch. Instead, it will automatically redeploy whenever a release is tagged in GitHub.
+
+The staging web GUI will track the `staging` branch in `dandiarchive` and will automatically redeploy whenever `staging` is changed.
+
+The production web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
+
+* A developer has a feature they want to add that impacts both the backend and the frontend.
+* They checkout a new branch in `dandi-api` and `dandiarchive`, do the work, and submit related PRs in both repos.
+* Once approved, the `dandi-api` PR is merged into `master` and the `dandiarchive` PR is merged into `staging`.
+* The changes are tested in staging.
+* Once approved, a new release of `dandi-api` is tagged, pushing the changes to production.
+* Once approved, the developer manually merges `dandiarchive/staging` into `dandiarchive/master`.
+
+## TODO
+Figure out a less manual release procedure for `dandiarchive`.
```

Comment:
perhaps once merged into staging, an action can autocreate a PR to merge staging into master.

Reply:
We use intuit's `auto` for dandi-cli, dandischema, and others. Could be used here as well.

Reply:
As I mentioned above, I think leaving the dandiarchive workflow largely alone could also work, perhaps with a bit of CI that checks to see whether `netlify.toml` has been modified and disallows merge to master if it has.